corehq/apps/case_search/xpath_functions/query_functions.py

```diff
@@ -92,7 +92,8 @@ def fuzzy_match(node, context):
     property_name = _property_name_to_string(node.args[0], node)
     value = unwrap_value(node.args[1], context)
 
-    return case_property_query(property_name, value, fuzzy=True)
+    return case_property_query(property_name, value, fuzzy=True,
+                               fuzzy_prefix_length=context.config.fuzzy_prefix_length)
```

Comment:
At present this is the only place I've set the fuzzy prefix length.  It doesn't (yet) apply to top level fuzzy parameters, although it probably should, at least when the fuzziness comes from the config.

There's a bit of weirdness there, where the `CaseSeachConfig` lets you set some properties as fuzzy, which is applied when those are _normal_ properties filtered at the top level, but this is ignored in the CSQL.  Instead, there's the `fuzzy-match` function in the DSL that allows for more targeted fuzziness.  Then there's a `fuzzy` arg accepted by the filter DSL entry point, which applies to ALL equality operations in the DSL.  I think this is used primarily for the pattern where we assemble a CSQL expression in python to do a parent property lookup, like `parent/foo = "bar"`

I'd welcome some reconciliation there, like getting rid of the "fuzzy" option in the DSL, requiring use of `fuzzy-match` explicitly.  Maybe also using a utility function for ancestor lookups, rather than expressing it as a CSQL string and running it through  our parser.

Reply:
update: I remembered that the fuzzy prefix actually does apply to the top level search parameters that go through the xpath utils (ie ancestor queries), so I figured I'd add it for all top level search parameters for consistency.  Pushing that commit now, so I think we're all set there (though I think there's room to make the code clearer if any reviewers have ideas and motivation).